src/components/LayersPanel/LayersPanel.js

```diff
@@ -27,6 +27,23 @@ class LayersPanel extends React.PureComponent {
       window.docViewer.updateView();
     }
   }
+  componentDidMount() {
+    core.addEventListener('layersUpdated', this.onLayersUpdated.bind(this));
+  }
+
+  componentWillUnmount() {
+    core.removeEventListener('layersUpdated', this.onLayersUpdated.bind(this));
+  }
+
+  onLayersUpdated() {
+    const doc = core.getDocument();
+    doc.getLayersArray().then(layers => {
+      if (layers.length) {
+        this.props.setLayers(layers);
+        this.forceUpdate();
```

Comment:
Can you explain why we need a `forceUpdate` here?

Reply:
Because change is coming from backend, UI is not updating, so we need to force update

Reply:
hmm ok, so does `this.props.setLayers(layers);` actually do anything? Maybe we can add a comment explaining why `this.forceUpdate` is needed here.

Reply:
setLayers is updating parent component when layers change we supposed to update parent like in line 69.